src/server.rs

```diff
@@ -69,6 +71,84 @@ fn replace_string_in_range(
     contents
 }
 
+fn function_defines(name: String, f: &FunctionExpr) -> bool {
+    for param in f.params.clone() {
```

Comment:
We only need `f` to get to `f.params`. Perhaps this function should just take the `params`, and leave the cloning or borrow to the callee.

Reply:
I went to look at what the type for `params` is: it's a `Vec<FunctionParameter>`. That means this function could just take a borrow, no clone needed.

I'm not sure if this is relevant here, but idiomatic rust seems to shy away from using `for` loops. I _think_ is probably okay, because this method preempts the loop when a match is found, whereas using an iterator means you'll have to iterate the whole vector, collect and assert on the length.